lua/telescope/previewers.lua

```diff
@@ -236,11 +280,46 @@ previewers.new_termopen_previewer = function(opts)
       cwd = opts.cwd or vim.fn.getcwd(),
     }
 
+    -- TODO(conni2461): Workaround for neovim/neovim#11751.
+    -- If upstream bug is fixed replace with term_opts.env = env
+    local gen_cmd = function(status)
+      local concat_env = function()
+        local env = {}
+        for k, v in pairs(termopen_env) do
+          table.insert(env, k .. '=' .. v)
+        end
+        return env
+      end
+
+      local shell = vim.o.shell
+      if shell == 'cmd.exe' then
+        local env = concat_env()
+        -- Based on https://superuser.com/questions/223104
+        return 'cmd /V /C "set ' ..
+               table.concat(env, '&& ') .. '&& ' ..
+               table.concat(opts.get_command(entry, status), ' ') .. '"'
+      elseif shell == 'powershell.exe' then
+        -- Who even uses powershell 😢
+        -- Based on https://stackoverflow.com/a/62627205
+        local output = ''
+        for k, v in pairs(termopen_env) do
+          output = output .. "$env:" .. k .. "='" .. v .. "'; "
+        end
+        output = output .. table.concat(opts.get_command(entry, status), ' ') .. '; '
+        for k, _ in pairs(termopen_env) do
+          output = output .. "$env: " .. k .. "=''; "
+        end
+        return output
+      else
+        local env = concat_env()
+        return table.concat(env, ' ') .. ' ' ..
+               table.concat(opts.get_command(entry, status), ' ')
+      end
+    end
```

Comment:
I closed the old conv. This is the new "shit".
It already worked with `bash`, `zsh`, and `dash`.
Handles `cmd.exe` and `powershell.exe`, if these are actually the correct names :laughing:. Are there any other shells i am currently not thinking about?
My implementation is based on these two threads. And i printed the resulting outputs to check if everything works theoretically but i can't test it.
Its a super bad workaround but it would be great if we could manage to get this working for all platforms.

Reply:
Just testing inside a windows vm. Both not yet working.

Reply:
So it works for fd with both `cmd.exe` and `powershell.exe`.
For `git_commit` etc preview does not work with `set shell=cmd.exe`. It fails with `terminal is not fully functional`. Does also not work on `master` so i guess that is a upstream/cmd.exe bug?! Just running something like `lua vim.fn.termopen({'git', 'log'})` will also give that result. So i won't bother as long as no windows user will report it :laughing: 

With `set shell=powershell.exe` git does not work at all. Just running `!git` will result in `An unknown error was received. This can happen if the remote process closed or ended abnormally. shell returned 4000`. Seems like an typical windows error. Maybe i did something wrong. When setting up all this stuff inside a vm.

Reply:
After I slept over it for a night, I think it's stupid. We should just not support env on powershell and cmd for now. And write something like this.
```lua
local get_cmd = function(status)
  if vim.o.shell == 'powershell.exe' or vim.o.shell == 'cmd.exe' then -- we should probably do a string.find here
    if table.getn(config.values.set_env) ~= 0 then
      print('Env is currently not supported for your shell')
    end
    return opts.get_command(entry, status)
  else
    return table.concat(env, ' ') .. ' ' .. table.concat(opts.get_command(entry, status), ' ')
  end
end
```

Reply:
Instead of printing, let's silently ignore and make a note in the readme. No need to spam people with the same message every time. (we can of course still pass the dict into `env`, just in case that ever gets fixed in master)

I am fine with just making sure that we do not attempt to run for invalid shells. That is good with me.

TY for all the investigation.